internal/tracer/exporter.go

```diff
@@ -0,0 +1,129 @@
+package tracer
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/windmilleng/wmclient/pkg/dirs"
+	exporttrace "go.opentelemetry.io/otel/sdk/export/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+
+	"github.com/windmilleng/tilt/pkg/logger"
+)
+
+type Locker sync.Locker
+
+type exporter struct {
+	outgoingMu sync.Mutex
+	dir        *dirs.WindmillDir
+
+	// members for communicating with the loop() goroutine
+	spanDataCh chan *exporttrace.SpanData
+	stopCh     chan struct{}
+
+	// members only accessed by the loop() goroutine
+	queue      []*exporttrace.SpanData
+	needsFlush bool
+}
+
+// TODO(dbentley): write tests for this
+func newExporter(ctx context.Context, dir *dirs.WindmillDir) (*exporter, error) {
+	r := &exporter{
+		dir:        dir,
+		spanDataCh: make(chan *exporttrace.SpanData),
+		stopCh:     make(chan struct{}),
+	}
+	go r.loop(ctx)
+	return r, nil
+}
+
+const OutgoingFilename = "usage/outgoing.json"
+
+func (e *exporter) loop(ctx context.Context) {
+	var flushingCh chan struct{}
+	var timerCh <-chan time.Time
+	for {
+		select {
+		case sd := <-e.spanDataCh:
+			e.queue = append(e.queue, sd)
+			if timerCh == nil {
+				timerCh = time.NewTimer(5 * time.Second).C
```

Comment:
this use of timer isn't quite right, see the comments on 
https://golang.org/pkg/time/#After

usually, you'd either keep the Timer around (and call Stop() when you don't need it anymore), or you'd use time.After()

have you considered keeping a single Timer outside of the loop that fires every 5 sec and does a flush()? this throttling algorithm feels a lot more complicated than it needs to be at this point.